### PR TITLE
Add global filter

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- #4
 
## Changes
- Add global filter for table and map

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/Home/Filters/index.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    countries?: NonNullable<CountryListQuery['public']['allCountries']>;
+    urgencyList?: NonNullable<AlertEnumsQuery['enums']['AlertInfoUrgency']>;
+    severityList?: NonNullable<AlertEnumsQuery['enums']['AlertInfoSeverity']>;
+    certaintyList?: NonNullable<AlertEnumsQuery['enums']['AlertInfoCertainty']>;
+}
+
+function Filters(props: Props) {
+    const {
+        value,
+        onChange,
```
Comment (puranban) [inline src/views/Home/Filters/index.tsx]:
```suggestion
        onValueChange,
```

File: src/views/Home/index.tsx [hunk 2]
```diff
@@ -8,18 +15,83 @@ import {
...
+      iso3
+      filteredAlertCount
+    }
+  }
+}
+`;
+
+const defaultFilterValue: FilterValue = {
+    countries: [],
+    regions: [],
```
Comment (tnagorra) [inline src/views/Home/index.tsx]: Should we allow multiple countries and multiple regions filter? @thenav56 @frozenhelium @samshara

File: src/views/Home/Filters/index.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+interface AlertFilters {
+    key: string;
+    label: string;
+}
+
+const countryKeySelector = (country: CountryType) => country?.id;
+
+const keySelector = (alert: AlertFilters) => alert?.key;
+const labelSelector = (alert: AlertFilters) => alert?.label;
```
Comment (tnagorra) [inline src/views/Home/Filters/index.tsx]:
```suggestion
const keySelector = (alert: AlertFilters) => alert.key;
const labelSelector = (alert: AlertFilters) => alert.label;
```

File: src/views/Home/Filters/index.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import styles from './styles.module.css';
+
+type CountryType = NonNullable<NonNullable<CountryListQuery['public']>['allCountries']>[number];
+
+interface AlertFilters {
+    key: string;
+    label: string;
+}
+
+const countryKeySelector = (country: CountryType) => country?.id;
```
Comment (tnagorra) [inline src/views/Home/Filters/index.tsx]:
```suggestion
const countryKeySelector = (country: CountryType) => country.id;
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by samshara — APPROVED** [review]

**Review by puranban — CHANGES_REQUESTED** [review]

File: src/views/Home/index.tsx [hunk 5]
```diff
@@ -8,18 +15,82 @@ import {
...
+      AlertInfoUrgency {
+        label
+        key
+      }
+      AlertInfoSeverity {
+        key
+        label
+      }
+    }
+}`;
```
Comment (puranban) [inline src/views/Home/index.tsx:56]:
```suggestion
const ALERT_ENUMS = gql`
    query AlertEnums {
        enums {
          AlertInfoCertainty {
            key
            label
          }
          AlertInfoUrgency {
            label
            key
          }
          AlertInfoSeverity {
            key
            label
          }
        }
    }
`;
```

File: src/views/Home/index.tsx (hunk 2 above)
Comment (tnagorra) [reply src/views/Home/index.tsx]: @barshathakuri Why was this conversation resolved? Did we change them to single select?

File: src/views/Home/Filters/index.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    urgencyList: string[];
+    severityList: string[];
+    certaintyList: string[];
+}
+
+interface Props {
+    value: FilterValue;
+    onChange: React.Dispatch<React.SetStateAction<FilterValue>>;
+    admin1?: NonNullable<NonNullable<NonNullable<AdminListQuery['public']>['admin1s']>['items']>;
+    countries?: NonNullable<CountryListQuery['public']['allCountries']>;
```
Comment (puranban) [inline src/views/Home/Filters/index.tsx]: use the type defined above for `countries`, `admin1`

File: src/views/Home/Filters/index.tsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        value.countries,
+        admin1,
+        countries,
+    ]);
+
+    return (
+        <div className={styles.filters}>
+            <SelectInput
+                placeholder={strings.alertCountries}
+                name="countries"
```
Comment (puranban) [inline src/views/Home/Filters/index.tsx]:
```suggestion
                name="country"
```

**Review by puranban — DISMISSED** [review]

**Review by frozenhelium — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**